packages/lib/test/src/helpers/encodeCall.test.js

```diff
@@ -0,0 +1,192 @@
+'use strict';
+
+require('../../setup');
+
+import encodeCall, { decodeCall } from '../../../src/helpers/encodeCall';
+import BN from 'bignumber.js';
+import _ from 'lodash';
+
+const NAME = 'myFunction';
+
+function assertGoodEncoding(types, values) {
+  const encoded = encodeCall(NAME, types, values).substring(10); // Remove signature hash.
+  const decoded = decodeCall(types, `0x${encoded}`);
+  if(values.length !== decoded.length) throw new Error('Invalid ecoding/decoding: Mismatch in number of encoded and decoded values.');
+  _.zipWith(values, decoded, (value, decodedValue) => {
+    if(Buffer.isBuffer(value)) value = `0x${value.toString('hex')}`;
+    if(value.toString() != decodedValue.toString()) throw new Error(`Invalid encoding/decoding. Encoded: ${value}, Decoded: ${decodedValue}`);
+  });
+}
+
+function assertBadEncoding(types, values, errorRegex) {
+  expect(() => encodeCall(NAME, types, values)).to.throw(errorRegex);
+}
```

Comment:
I'd move the assert functions to the end of the file to ensure that the first thing you see when opening a test file is the `describe` statement